### ISSUE TITLE
SEPA v2: self-supervised λ schedule from execution entropy variance

### ISSUE THREAD

**Issue by teilomillet** [teilomillet/textpolicy#64]
## Context

PR #63 implements SEPA with a linear λ schedule (`sepa_steps=N`, λ = min(step/N, 1.0)). This works but requires picking N upfront — a value that depends on model size, task distribution, and how fast execution consolidates. The linear schedule doesn't know when Phase ① is actually ending.

## Proposal

Replace the fixed schedule with a self-supervised one: derive λ from the model's own execution token entropy variance. When variance drops, execution has consolidated and λ should increase. The model signals when it's ready for the gate.

### State

`_GTPOTransform` gains three persistent fields:

- `_var_ema: float` — exponential moving average of execution entropy variance
- `_var_0: Optional[float]` — initial variance (captured from first batch, used as baseline)
- `_batch_count: int` — tracks initialization

### Per-batch computation (inside `__call__`, after computing `exec_mask`)

```python
exec_H = flat_entropies * exec_mask
exec_count = mx.sum(exec_mask)
mean_H = mx.sum(exec_H) / mx.maximum(exec_count, 1.0)
var_batch = mx.sum(exec_mask * (flat_entropies - mean_H) ** 2) / mx.maximum(exec_count, 1.0)
var_batch = var_batch.item()  # scalar, outside gradient graph

if self._var_0 is None:
    self._var_0 = var_batch
    self._var_ema = var_batch
else:
    self._var_ema = decay * self._var_ema + (1 - decay) * var_batch

ratio = self._var_ema / max(self._var_0, eps)
lambda_t = 1.0 - min(ratio / threshold, 1.0)
```

### λ derivation

```
ratio = var_ema / var_0          # how much variance remains vs. initial
lambda_t = 1 - clamp(ratio / threshold, 0, 1)
```

| `ratio` | `λ` (threshold=0.2) | Interpretation |
|---|---|---|
| 1.0 (start) | 0.0 | Full variance → pure GTPO |
| 0.5 | 0.0 | Still above threshold → no pooling |
| 0.2 | 0.0 | At threshold → pooling begins |
| 0.1 | 0.5 | Below threshold → partial pooling |
| 0.0 | 1.0 | Zero variance → full pooling |

Correction: with `1 - clamp(ratio / threshold, 0, 1)`:

| `ratio` | `ratio / 0.2` | `clamp` | `λ` |
|---|---|---|---|
| 1.0 | 5.0 | 1.0 | 0.0 |
| 0.4 | 2.0 | 1.0 | 0.0 |
| 0.2 | 1.0 | 1.0 | 0.0 |
| 0.1 | 0.5 | 0.5 | 0.5 |
| 0.0 | 0.0 | 0.0 | 1.0 |

λ stays at 0 until variance drops below `threshold × var_0`, then ramps linearly to 1 as variance approaches 0.

### Bidirectionality

If execution entropy variance spikes (e.g. new task distribution, curriculum shift), `ratio` increases, `λ` drops, and the model re-enters Phase ① naturally. No manual intervention.

### Parameters

Two new parameters (or defaults that don't need tuning):

- **`sepa_ema_decay: float = 0.99`** — EMA smoothing. 0.99 ≈ 100-batch effective window. Higher = more stable, slower to react.
- **`sepa_var_threshold: float = 0.2`** — fraction of initial variance at which λ starts rising. 0.2 = "anneal when variance drops to 20% of initial."

Both are relative/normalized, so they should transfer across model sizes and tasks without tuning.

### API

```python
# v2: self-supervised schedule
transform = build_gtpo_transform(
    alpha_1=1.0, alpha_2=0.1,
    tokenizer=tokenizer,
    sepa_schedule="auto",           # NEW — selects self-supervised
    sepa_ema_decay=0.99,            # optional, default 0.99
    sepa_var_threshold=0.2,         # optional, default 0.2
)

# v1: linear schedule (unchanged)
transform = build_gtpo_transform(
    alpha_1=1.0, alpha_2=0.1,
    tokenizer=tokenizer,
    sepa_steps=500,
)
```

`sepa_schedule="auto"` and `sepa_steps > 0` are mutually exclusive. If both are set, raise ValueError.

### mx.compile concern

`var_batch.item()` pulls a scalar out of the MLX graph. This is the same pattern `boost_entropy_with_planning` avoids via `mx.stop_gradient`. However, `_var_ema` is used to compute `lambda_t` (a Python float), which enters the graph as a constant — it doesn't create a gradient path. Verify this doesn't break `mx.compile` tracing. If it does, the fallback is to compute `lambda_t` in `prepare_batch` (which runs outside compile) and store it in `batch_data["sepa_lambda"]`.

### State and serialization

The transform becomes stateful (`_var_ema`, `_var_0`). Considerations:
- Checkpointing: should these be saved/restored? Probably yes for training resumption.
- The Trainer doesn't currently serialize transform state. May need a `state_dict()`/`load_state_dict()` protocol.
- For now, acceptable to lose this state on restart (λ re-calibrates from the resumed model's variance, which is arguably more correct than restoring a stale baseline).

## Acceptance criteria

- [ ] `sepa_schedule="auto"` activates self-supervised λ
- [ ] λ=0 at start (first batch), increases as execution entropy variance drops
- [ ] λ can decrease if variance spikes (bidirectional)
- [ ] `sepa_schedule="auto"` + `sepa_steps > 0` raises ValueError
- [ ] Works under `mx.compile` (no tracing errors)
- [ ] Existing linear schedule (`sepa_steps`) unaffected
- [ ] Test: synthetic variance drop → λ increases
- [ ] Test: synthetic variance spike → λ decreases

Depends on #63.
